### ISSUE TITLE
const-correctness improvement suggestion

### ISSUE THREAD

**Issue by MaciejPatro** [p-ranav/argparse#154]
Hello,

```cpp
  /* Getter enabled for all template types other than std::vector and std::list   
   * @throws std::logic_error in case of an invalid argument name                 
   * @throws std::logic_error in case of incompatible types                       
   */                                                                             
  template <typename T = std::string> T get(std::string_view aArgumentName) {     
    return (*this)[aArgumentName].get<T>();                                       
  }                                                                               
```

Function is providing a copy of value - but is not const - which leaks this to other classes that have `ArgumentParser` as a member.

Example of problem:
```cpp
struct program_arguments
{
  int age()  /* const is not possible here */ { return parser.get<int>("age"); }
  argparse::ArgumentParser parser;
};
```

There seems to be no reason besides usage of `[] operator` for the  function not to be const. It could be made `const` for example by using `.at` on `std::map`

**Comment by skrobinson** [comment]
Was this addressed in #74 previously? Or have you found a new issue?

**Comment by MaciejPatro** [comment]
Yes, you are fully right - sorry for the confusion!
